[DOC] Switch from Sphinx to MkDocs

ericmjl opened the pull request:
This resolves #894. The majority of votes from active maintainers is to move to mkdocs.

The things done in this PR include:

1. Switching our docs build system from sphinx to mkdocs.
2. Refactoring `functions.py` and `utils.py` to play more nicely with automatic API docs.
3. Remove notebooks from the docs (temporarily) until we bring them back.
4. Mark tests as being `turtle` (slow), in preparation for splitting tests into two to make them run faster.

ericmjl committed:
First commit test-driving mkdocs

Some thoughts left for later:

1. It's going to be complicated to migrate.
2. We'd probably want to scope out the amount of effort needed
to migrate the docs in its entirety
before deciding whether we want to do so or not.
3. Probably best to write out a migration plan before embarking.

ericmjl committed:
Merge branch 'dev' into mkdocs

ericmjl committed:
Add mkdocs-related packages to dev environment spec

ericmjl committed:
Add CSS styling

ericmjl committed:
More fixes.

ericmjl committed:
Add doc references

ericmjl committed:
Update change authors to Markdown.

ericmjl committed:
Added in a whole bunch of API docs

ericmjl committed:
Add alias to docs README.

ericmjl committed:
Update URLs on docs index page

ericmjl committed:
Switched out docs build

samukweku commented:
@ericmjl how can I help?

ericmjl commented:
> @ericmjl how can I help?  

Thank you, @samukweku! The biggest help I think I can get here is to reformat docstrings such that docstrings are formatted as Markdown style while leaving the parameter descriptions alone. 

This is going to be a slow push to make it happen; the activity involved, which is hand-checking each function one at a time, is fundamentally kinda boring and repetitive, so we should take it slow and easy here.

loganthomas commented:
@ericmjl I’d like to help as well! Is there a way we can divvy up the sub modules as to not duplicate effort?

loganthomas commented on mkdocs/api/chemistry.md:
```diff
@@ -0,0 +1,3 @@
+# Biology
```
@ericmjl this should be chemistry

loganthomas commented on mkdocs/api/timeseries.md:
```diff
@@ -0,0 +1,3 @@
+# PySpark
```
@ericmjl this should be timeseries

ericmjl commented:
@loganthomas and @samukweku thanks! Ok, tomorrow evening I will list out the submodules to review (will edit my original post) and then let you guys claim what you think you'd like to start off with.

ericmjl committed:
Fix names h/t @loganthomas

ericmjl committed:
Add script to count number of functions defined per .py file

ericmjl committed:
Find/replace on backticks

nvamsikrishna05 commented:
@ericmjl  Let me know where I can help.

ericmjl committed:
Pin minimum versions of packages.

ericmjl committed:
Make dev env use mamba to install packages.

ericmjl committed:
Add mkdocs packages to environment

ericmjl committed:
Fix biology docstring

ericmjl committed:
Update devcontainer spec

ericmjl commented:
@loganthomas and @samukweku I updated my original post! `functions.py` has the most meat in there. We might need to divvy that one up.

ericmjl committed:
Update devcontaienr

ericmjl committed:
Flake8 fixes

nvamsikrishna05 commented:
@ericmjl  Thanks for setting it up. 

@loganthomas  @samukweku  I'll pickup the `finance.py` file as a starting point.

loganthomas commented on CONTRIBUTING.rst:
```diff
@@ -34,7 +34,7 @@ To get started:
     click on the quick actions Status Bar item in the lower left corner.
 5. Then select "Remote Containers: Clone Repository In Container Volume".
-6. Enter in the URL of your fork of ``pyjanitor``.
+6. Enter in the URL of your fork of `pyjanitor``.
```
Hanging backtick

loganthomas commented on CONTRIBUTING.rst:
```diff
@@ -73,8 +73,8 @@ To run correctly inside the environment,
-    you may need to install ``make`` before you can run the install.
-    You can get it from ``conda-forge``::
+    you may need to install `make` before you can run the install.
+    You can get it from `conda-forge``::
```
Hanging backtick

loganthomas commented on CONTRIBUTING.rst:
```diff
@@ -86,7 +86,7 @@ make sure you select the correct kernel from the top right corner of JupyterLab!
 
 As of 29 October 2019,
-``pre-commit`` hooks are available to run code formatting checks automagically before git commits happen.
+``pre-commit` hooks are available to run code formatting checks automagically before git commits happen.
```
Hanging backtick

loganthomas commented on CONTRIBUTING.rst:
```diff
@@ -96,18 +96,18 @@ run the following commands::
 The command above allows you to view the documentation locally in your browser.
 `Sphinx (a python documentation generator) <http://www.sphinx-doc.org/en/stable/usage/quickstart.html>`_ builds and renders the html for you,
-and you can find the html files by navigating to ``pyjanitor/docs/_build``,
+and you can find the html files by navigating to `pyjanitor/docs/_build``,
```
Hanging backtick

loganthomas commented on CONTRIBUTING.rst:
```diff
@@ -96,18 +96,18 @@ run the following commands::
+open the `index.html` file.
 
-.. note:: If you get any errors related to Importing modules when running ``make docs``,
+.. note:: If you get any errors related to Importing modules when running `make docs``,
```
Hanging backtick

loganthomas commented on CONTRIBUTING.rst:
```diff
@@ -146,19 +146,19 @@ Now you can make your changes locally.
-* Code unit testing: ``make test``
+* Automated code formatting: `make style``
+* Code styling problems check: `make lint``
+* Code unit testing: `make test``
```
Hanging backticks on 3 lines above

loganthomas commented on CONTRIBUTING.rst:
```diff
@@ -146,19 +146,19 @@ Now you can make your changes locally.
 Styling problems must be resolved before the pull request can be accepted.
 
-``make test`` runs all ``pyjanitor``'s unit tests to probe whether changes to the source code have potentially introduced bugs.
+``make test` runs all `pyjanitor``'s unit tests to probe whether changes to the source code have potentially introduced bugs.
```
Hanging backtick

loganthomas committed:
[DOC] MkDocs Machine Learning (#900)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* [DOC] mkdocs updates to ml module

* [DOC] mkdocs updates to ml module

* [DOC] mkdocs updates to ml module

* [DOC] mkdocs updates to ml module; black formatting

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

loganthomas committed:
[DOC] MkDocs IO (#903)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* [DOC] mkdocs updates to read_csv() in io

* [DOC] mkdocs updates to incldue io in docs

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

loganthomas committed:
[DOC] MkDocs Math (#901)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* [DOC] mkdocs updates to log() in math

* [DOC] mkdocs updates to ecdf() in math

* [DOC] mkdocs updates to exp() in math

* [DOC] mkdocs updates to logit() in math

* [DOC] mkdocs updates to normal_cdf() in math

* [DOC] mkdocs updates to probit() in math

* [DOC] mkdocs updates to single line docstrings in math

* [DOC] mkdocs updates to z_score() in math

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

ericmjl committed:
[DOC] MkDocs Chemistry (#899)

* mkdocstrings on chemistry functions

* Minor update to biology.py

* Proper config for syntax highlighting

* Switch to nord theme for syntax highlighting

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* FLake8 fix.

Co-authored-by: Zero <Zeroto521@gmail.com>

loganthomas committed:
[DOC] MkDocs cleanup (#908)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* [CLN] remove README.rst in favor of mkdocs/index.md

* [CLN] remove CONTRIBUTING.rst in favor of mkdocs/devguide.md

* [DOC] add symlink to mkdocs/devguide.md for CONTRIBUTING.md

* [FIX] remove file CONTRIBUTING.md

* [DOC] add symlink to mkdocs/devguide.md for CONTRIBUTING.md

* [INF] update setup.py to replace README.rst with mkdocs/index.md

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

loganthomas committed:
[DOC] MkDocs TimeSeries (#905)

* [DOC] mkdocs updates to fill_missing_timestamps() in timeseries

* [DOC] mkdocs updates to  _get_missing_timestamps() in timeseries

* [DOC] mkdocs updates to  sort_timestamps_monotonically() in timeseries

* [DOC] mkdocs updates to flag_jumps() in timeseries

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

loganthomas committed:
[DOC] MkDocs AUTHORS (#907)

* [DOC] mkdocs updates to AUTHORS

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

loganthomas committed:
[DOC] MkDocs Engineering (#904)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* [DOC] mkdocs updates to convert_units() in engineering

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

nvamsikrishna05 committed:
MKdocs functions - 1 (#914)

nvamsikrishna05 committed:
Mkdocs functions -2 (#915)

loganthomas committed:
[DOC] MkDocs Utils Part 2 (#913)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* [DOC] mkdocs updates to refactored_function() in utils

* [DOC] mkdocs updates to rename_kwargs() in utils

* [DOC] mkdocs updates to check_column() in utils

* [DOC] mkdocs updates to skipna() in utils

* [DOC] mkdocs updates to skiperror() in utils

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

loganthomas committed:
[DOC] MkDocs Utils Part 1 (#912)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* [DOC] mkdocs updates to check() in utils

* [DOC] mkdocs updates to _strip_underscores() in utils

* [DOC] mkdocs updates to import_message() in utils

* [DOC] mkdocs updates to idempotent() in utils

* [DOC] mkdocs updates to deprecated_alias() in utils

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

ericmjl committed:
Add mknotebooks to dev environment

loganthomas committed:
[DOC] MkDocs Utils Part 4 (#923)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* DOC: update _sub_expand_grid(pd.DataFrame) in utils

* DOC: update _sub_expand_grid(pd.Index) in utils

* DOC: update _data_checks_complete() in utils

* DOC: update  _computations_complete() in utils

* DOC: update _create_indexer_for_complete() in utils

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

loganthomas committed:
[DOC] MkDocs Utils Part 3 (#922)

* [INF] simplify a bit linting, use pre-commit as CI linting checker (#892)

* [INF] simplify a bit linting

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.

We should use one of them as the standard, not both.

* Update CHANGELOG.md

* should be `--config` not `-c`

`-c` = `--code TEXT`, Format the code passed in as a string.

* remove nbstripout

keep the same with the old code-checks.yml

* lint via pre-commit

* Update CHANGELOG.md

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* DOC: update _computations_expand_grid() in utils

* DOC: update _expand_grid() in utils

* DOC: update _sub_expand_grid() in utils

* DOC: update _sub_expand_grid(np.ndarray) in utils

* DOC: update _sub_expand_grid(pd.Series) in utils

* [DOC] fix NumPy

Co-authored-by: Zero <Zeroto521@gmail.com>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

nvamsikrishna05 committed:
[DOC] MkDocs Finance (#909)

* MkDocs Finance

* Update janitor/finance.py

* Flake8 fix on finance.py

* mkdocs finance fixes

* finance function duplicate line fix

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

ericmjl committed:
[MKDOCS] MkDocs Functions (#920)

* [DOC] First pass attempt at cleaning up the functions.py module

* [MKDOCS] mkdocs collapse_levels, bin_num, ... (#919)

* mkdocs collapse_levels, bin_num, ...

* updates

* mkdocs updates (#918)

* [MKDOCS] Mkdown pivot longer groupby topk complete expand grid (#917)

* updates

* updates

* updates

* updates

* updates

* mkdocs filters (#921)

* mkdocs_modify_values (#924)

* [DOCS] Mkdocs function - 3 (#925)

* COnditional join docstring formatting fixes

* pivot wider docstring formatting fixes

* Update janitor/functions.py

* Update janitor/functions.py

Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

* Reinstate DAR402.

Co-authored-by: Samuel Oranyeli <samueloranyeli@gmail.com>
Co-authored-by: Vamsi Krishna <nvamsikrishna05@users.noreply.github.com>

ericmjl committed:
Merge remote-tracking branch 'origin/dev' into mkdocs

ericmjl committed:
Update docstring to be correct.

ericmjl commented:
Next steps to note:

1. Jupyter notebooks, in this scheme, get moved out to another repo, so I am going to disable Jupyter nb builds.
2. Need docs previews to make sure everything is running properly.

ericmjl committed:
Ensured pivot data from wide to long NB works.

ericmjl committed:
Doc fixes.

ericmjl committed:
Merge branch 'dev' into mkdocs

ericmjl committed:
Add quickstart to README

ericmjl committed:
Add netlify docs preview build

ericmjl committed:
Official docs build on dev branch

ericmjl committed:
Refactor environment build scripts

ericmjl commented:
<!-- NETLIFY DEPLOY COMMENT GENERATED BY ACTIONS_NETLIFY -->
🚀 Deployed on https://deploy-preview-897--pyjanitor.netlify.app

ericmjl committed:
Merge remote-tracking branch 'origin/mkdocs' into mkdocs

ericmjl committed:
Merge remote-tracking branch 'origin/dev' into mkdocs

samukweku commented on CHANGELOG.md:
```diff
@@ -2,16 +2,21 @@
+-   [ENH] `expand_grid` returns a MultiIndex DataFrame, allowing the user to decide how to manipulate the columns. @samukweku
+-   [INF] Simplify a bit linting, use pre-commit as the CI linting checker. @Zeroto521
+-   [ENH] Fix bug in `pivot_longer` for wrong output when `names_pattern` is a sequence with a single value. Issue #885 @samukweku
+-   [ENH] Deprecate `aggfunc` from `pivot_wider`; aggregation can be chained with pandas' `groupby`.
```
I think line 13 is a repeat

ericmjl committed:
Merge branch 'dev' into mkdocs

ericmjl committed:
Merge remote-tracking branch 'origin/dev' into mkdocs

ericmjl committed:
Begin refactoring functions module

ericmjl committed:
Massive refactor of functions into their individual source files.

ericmjl committed:
Consolidate filter functions into filter.py

ericmjl committed:
Import all of the original functions back into top-level namespace.

ericmjl committed:
Fix imports for docs build

ericmjl committed:
Make JoinOperator and JoinTypes private so that they don't show up on docs

ericmjl committed:
Move data description accessor to accessors submodule

ericmjl committed:
Add module docstring for clean_names.py

ericmjl committed:
Make FILLTYPE private show that it doesn't show up on docs

ericmjl committed:
Switch import of DataDescription into janitor.accessors.

ericmjl committed:
Unify fill functions into fill.py

ericmjl committed:
Reinstate functions API docs

ericmjl committed:
Ready to flip the switch

ericmjl committed:
Make CategoryOrder private to prevent it from showing up on API docs

ericmjl committed:
Make as_categorical_checks private to prevent it from showing up on API docs.

ericmjl committed:
Fix tests

ericmjl committed:
Fix tests

ericmjl committed:
CI trigger

ericmjl committed:
Marked a few tests as being turtle slow.

ericmjl committed:
Marked xfail test_dict_extension_array.

Also raised issue #952.

ericmjl committed:
Fix spark tests.

loganthomas commented on examples/convert_currency.md:
````diff
@@ -39,7 +39,7 @@ data_dict = {
 
 ## Example: Converting a column from one currency to another using rates from 01/01/2018
- ```python
+ ``python
````
```suggestion
 ```python
```

loganthomas commented on examples/filter_date.md:
````diff
@@ -70,7 +70,7 @@ example_dataframe = pd.DataFrame(date_list, columns=['AMOUNT', 'DATE'])
 
 ## Example 1: Filter dataframe between two dates
- ```python
+ ``python
````
```suggestion
 ```python
```

loganthomas commented on examples/filter_date.md:
````diff
@@ -84,7 +84,7 @@ example_dataframe.filter_date('DATE', start=start, end=end)
 
 ## Example 2: Using a different date format for filtering
- ```python
+ ``python
````
```suggestion
 ```python
```

loganthomas commented on examples/limit_column_characters.md:
````diff
@@ -32,7 +32,7 @@ data_dict = {
 
 ## Example1: Standard truncation
- ```python
+ ``python
````
```suggestion
 ```python
```

loganthomas commented on examples/round_to_fraction.md:
````diff
@@ -28,7 +28,7 @@ data_dict = {
 
 ## Example 1: Rounding the first column to the nearest half
- ```python
+ ``python
````
```suggestion
 ```python
```

loganthomas commented on examples/row_to_names.md:
````diff
@@ -41,7 +41,7 @@ data_dict = {
 
 ## Example1: Move first row to column names
- ```python
+ ``python
````
```suggestion
 ```python
```

loganthomas commented on examples/then.md:
````diff
@@ -26,7 +26,7 @@ data_dict = {
 
 ## Example 1
- ```python
+ ``python
````
```suggestion
 ```python
```

loganthomas commented on tests/xarray/conftest.py:
```diff
@@ -12,7 +12,7 @@ def da():
     that are either linearly or log-spaced increments.
 
-    Included is a simple metadata dictionary passed as ``attrs``.
+    Included is a simple metadata dictionary passed as `attrs``.
```
```suggestion
    Included is a simple metadata dictionary passed as `attrs`.
```

loganthomas commented on tests/functions/test_process_text.py:
```diff
@@ -193,7 +193,7 @@ def test_str_get():
 
 def test_str_lower():
-    """Test string conversion to lowercase using ``.str.lower()``."""
+    """Test string conversion to lowercase using `.str.lower()``."""
```
```suggestion
    """Test string conversion to lowercase using `.str.lower()`."""
```

loganthomas commented on tests/functions/test_pivot_wider.py:
```diff
@@ -45,7 +45,7 @@ def test_type_names_from(df_checks_output):
 
 def test_names_from_None(df_checks_output):
-    """Raise ValueError if no value is provided for ``names_from``."""
+    """Raise ValueError if no value is provided for `names_from``."""
```
```suggestion
    """Raise ValueError if no value is provided for `names_from`."""
```

loganthomas commented on tests/functions/test_pivot_longer.py:
```diff
@@ -319,7 +319,7 @@ def test_pivot_no_args_passed():
     """
     Test output if only `index` is passed and
-    `sort_by_apperance is ``True``.
+    `sort_by_apperance is `True``.
```
```suggestion
    `sort_by_apperance is `True`.
```

loganthomas commented on tests/functions/test_pivot_longer.py:
```diff
@@ -434,7 +434,7 @@ def test_pivot_index_patterns_only_sort_by_appearance(
     """
     Test output if the `patterns` function is passed to `index`,
-    and `sort_by_appearance` is ``True``.
+    and `sort_by_appearance` is `True``.
```
```suggestion
    and `sort_by_appearance` is `True`.
```

loganthomas commented on tests/chemistry/test_morgan_fingerprint.py:
```diff
@@ -34,7 +34,7 @@ def test_morgan_fingerprint_bits(chemdf):
 @pytest.mark.chemistry
 def test_morgan_fingerprint_kind_error(chemdf):
-    """Test ``morgan_fingerprint`` raises exception for invalid ``kind``."""
+    """Test `morgan_fingerprint` raises exception for invalid `kind``."""
```
```suggestion
    """Test `morgan_fingerprint` raises exception for invalid `kind`."""
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -23,19 +23,19 @@ def clone_using(
+    Given a NumPy array, return an XArray `DataArray` which contains the same
     dimension names and (optionally) coordinates and other properties as the
-    supplied ``DataArray``.
+    supplied `DataArray``.
```
```suggestion
    supplied `DataArray`.
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -23,19 +23,19 @@ def clone_using(
+    This is similar to `xr.DataArray.copy()` with more specificity for
     the type of cloning you would like to perform - the different properties
-    that you desire to mirror in the new ``DataArray``.
+    that you desire to mirror in the new `DataArray``.
```
```suggestion
    that you desire to mirror in the new `DataArray`.
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -51,24 +51,24 @@ def clone_using(
-        ``DataArray`` for the coordinates of the newly-generated array. Shapes
-        must match in this case. If ``False``, only the number of dimensions
+    :param da: The `DataArray` supplied by the method itself.
+    :param np_arr: The NumPy array which will be wrapped in a new `DataArray``
```
```suggestion
    :param np_arr: The NumPy array which will be wrapped in a new `DataArray`
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -51,24 +51,24 @@ def clone_using(
-        must match in this case. If ``False``, only the number of dimensions
+    :param da: The `DataArray` supplied by the method itself.
+    :param np_arr: The NumPy array which will be wrapped in a new `DataArray``
+        given the properties copied over from the source `DataArray``.
```
```suggestion
        given the properties copied over from the source `DataArray`.
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -51,24 +51,24 @@ def clone_using(
+    :param da: The `DataArray` supplied by the method itself.
+    :param np_arr: The NumPy array which will be wrapped in a new `DataArray``
+        given the properties copied over from the source `DataArray``.
+    :param use_coords: If `True``, use the coordinates of the source
```
```suggestion
    :param use_coords: If `True`, use the coordinates of the source
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -51,24 +51,24 @@ def clone_using(
+        given the properties copied over from the source `DataArray``.
+    :param use_coords: If `True``, use the coordinates of the source
+        `DataArray` for the coordinates of the newly-generated array. Shapes
+        must match in this case. If `False``, only the number of dimensions
```
```suggestion
        must match in this case. If `False`, only the number of dimensions
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -51,24 +51,24 @@ def clone_using(
-    :param use_attrs: If ``True``, copy over the ``attrs`` from the source
-        ``DataArray``.
-        The data inside ``attrs`` itself is not copied, only the mapping.
+    :param use_attrs: If `True``, copy over the `attrs` from the source
```
```suggestion
    :param use_attrs: If `True`, copy over the `attrs` from the source
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -51,24 +51,24 @@ def clone_using(
-        ``DataArray``.
-        The data inside ``attrs`` itself is not copied, only the mapping.
+    :param use_attrs: If `True``, copy over the `attrs` from the source
+        `DataArray``.
```
```suggestion
        `DataArray`.
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -51,24 +51,24 @@ def clone_using(
-    :param new_name: If set, use as the new name of the returned ``DataArray``.
-        Otherwise, use the name of ``da``.
-    :return: A ``DataArray`` styled like the input ``DataArray`` containing the
+    :param new_name: If set, use as the new name of the returned `DataArray``.
```
```suggestion
    :param new_name: If set, use as the new name of the returned `DataArray`.
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -51,24 +51,24 @@ def clone_using(
-    :raises ValueError: if shape of ``NumPy`` array and ``DataArray``
+    :raises ValueError: if number of dimensions in `NumPy` array and
+        `DataArray` do not match.
+    :raises ValueError: if shape of `NumPy` array and `DataArray``
```
```suggestion
    :raises ValueError: if shape of `NumPy` array and `DataArray`
```

loganthomas commented on janitor/xarray/functions.py:
```diff
@@ -107,8 +107,8 @@ def convert_datetime_to_number(
 
-    Usage example to convert a ``DataArray``'s time dimension coordinates from
-    a ``datetime`` to minutes:
+    Usage example to convert a `DataArray``'s time dimension coordinates from
```
```suggestion
    Usage example to convert a `DataArray`'s time dimension coordinates from
```

loganthomas commented on janitor/spark/backend.py:
```diff
@@ -13,7 +13,7 @@ class CachedAccessor:
     ----------
     name : str
-        The namespace this will be accessed under, e.g. ``df.foo``
+        The namespace this will be accessed under, e.g. `df.foo``
```
```suggestion
        The namespace this will be accessed under, e.g. `df.foo`
```

loganthomas commented on janitor/functions/fill.py:
```diff
@@ -0,0 +1,213 @@
+
+@pf.register_dataframe_method
+@deprecated_alias(columns="column_names")
+def fill_empty(
```
Should this be moved to `functions/fill_empty.py`?

ericmjl replied:
I consolidated both fill_direction and fill_empty into `fill.py`.

loganthomas reviewed (commented):
Looking good! Thanks for taking this on! 😍

ericmjl committed:
Update examples/convert_currency.md

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update examples/filter_date.md

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update examples/filter_date.md

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update examples/limit_column_characters.md

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update examples/round_to_fraction.md

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/spark/backend.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update examples/row_to_names.md

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update examples/then.md

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update tests/xarray/conftest.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update tests/functions/test_process_text.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update tests/functions/test_pivot_wider.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update tests/functions/test_pivot_longer.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update tests/functions/test_pivot_longer.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update tests/chemistry/test_morgan_fingerprint.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl committed:
Update janitor/xarray/functions.py

Co-authored-by: Logan Thomas <logan.thomas005@gmail.com>

ericmjl commented:
Note to self: Copy/paste the following into the squash commit message.
```
This resolves #894. The majority of votes from active maintainers is to move to mkdocs.

The things done in this PR include:

1. Switching our docs build system from sphinx to mkdocs.
2. Refactoring functions.py and utils.py to play more nicely with automatic API docs.
3. Remove notebooks from the docs (temporarily) until we bring them back.
4. Mark tests as being turtle (slow), in preparation for splitting tests into two to make them run faster.
```

samukweku commented:
`4. Mark tests as being turtle (slow), in preparation for splitting tests into two to make them run faster.
`

I totally missed that part
